mozjs-sys/src/jsglue.cpp

```diff
@@ -692,10 +692,26 @@ void DeleteCompileOptions(JS::ReadOnlyCompileOptions* aOpts) {
   delete static_cast<JS::OwningCompileOptions*>(aOpts);
 }
 
-JS::ReadOnlyCompileOptions* NewCompileOptions(JSContext* aCx, const char* aFile,
-                                              unsigned aLine) {
+enum class IntroductionType : unsigned {
+  Undefined,
+  InlineScript,
+};
+
+JS::ReadOnlyCompileOptions* NewCompileOptions(
+    JSContext* aCx, const char* aFile, unsigned aLine,
+    IntroductionType introductionType) {
```

Comment:
```suggestion
    const char* introductionType) {
```
I think we should just use here `const char*` thus let user of mozjs define custom values, because we will likely need more of them in servo eventually. And it also reduces amount of C++ code.

Also isn't this already available here: https://doc.servo.org/mozjs/jsapi/struct.TransitiveCompileOptions.html#structfield.introductionType

Reply:
actually given https://github.com/servo/servo/issues/38378 we should write safe wrapper for compile options and let them pass into evaluate_script.